sfge/src/main/java/com/salesforce/graph/ops/expander/ApexPathCollapserImpl.java

```diff
@@ -176,9 +198,12 @@ public void resultReturned(
             }
             // Clean up
             if (retained.isEmpty()) {
-                forkEventToApexExpandersWithResults.remove(forkEvent);
+                forkEventIdToApexExpanderIdsWithResults.remove(forkEvent.getId());
             } else {
-                forkEventToApexExpandersWithResults.put(forkEvent, retained);
+                forkEventIdToApexExpanderIdsWithResults.put(
+                        forkEvent.getId(),
+                        PathExpansionRegistryUtil.convertApexPathExpandersToIds(
+                                registry, retained));
```

Comment:
I tried to create [a collection type based on Indexable objects](https://github.com/forcedotcom/sfdx-scanner/tree/rm/oom-jan-23/pathRegInst/IndexableCollection/sfge/src/main/java/com/salesforce/graph/ops/registry) so that they wrap the overhead of converting back and forth from Ids. It started getting more tedious to handle the internal conversions and lot of inherent collection-things we take for granted. We can probably pick it up if/when we move another object into registry.